chcemvediet/apps/obligees/tests/test_forms.py

```diff
@@ -1,123 +1,175 @@
 # vim: expandtab
 # -*- coding: utf-8 -*-
-import unittest
-
+import lxml.html
 from django import forms
-from django.core.urlresolvers import reverse
 from django.core.exceptions import ValidationError
 from django.test import TestCase
+from django.utils.translation import ugettext_lazy as _
+
+from poleno.utils.urls import reverse
 
 from chcemvediet.tests import ChcemvedietTestCaseMixin
 
 from ..forms import ObligeeWidget, ObligeeField
 
 
-class ObligeeFieldWithTextInputWidgetTest(ChcemvedietTestCaseMixin, TestCase):
+class ObligeeFieldTest(ChcemvedietTestCaseMixin, TestCase):
     u"""
-    Tests ``ObligeeField`` with ``TextInput`` widget.
+    Tests ``ObligeeField`` field with ``ObligeeWidget`` widget.
     """
 
     class Form(forms.Form):
         obligee = ObligeeField()
 
     class FormWithWidgetAttrs(forms.Form):
         obligee = ObligeeField(
-                widget=forms.TextInput(attrs={
+                widget=ObligeeWidget(input_attrs={
                     u'class': u'custom-class',
                     u'custom-attribute': u'value',
                     }),
                 )
 
 
-    @unittest.skip(u'FIXME')
     def test_new_form(self):
         form = self.Form()
         rendered = self._render(u'{{ form }}', form=form)
+        html = lxml.html.fromstring(rendered)
+
         self.assertInHTML(u'<label for="id_obligee">Obligee:</label>', rendered)
+        root = html.get_element_by_id(u'id_obligee')
+        self.assertEqual(root.tag, u'div')
+        self.assertEqual(root.attrib[u'class'], u'chv-obligee-widget')
+
+        inputs = root.find_class(u'chv-obligee-widget-inputs')[0]
+        self.assertEqual(inputs.tag, u'div')
+        input = inputs.find_class(u'chv-obligee-widget-input')[0]
+        self.assertEqual(input.tag, u'div')
         self.assertInHTML(u"""
-                <input class="autocomplete" data-autocomplete-url="{url}" id="id_obligee" name="obligee" type="text">
+                <input class="form-control pln-autocomplete" data-autocomplete-url="{url}" name="obligee" data-name="obligee" type="text" value="">
                 """.format(url=reverse(u'obligees:autocomplete')), rendered)
```

Comment:
Preco toto assertujes cez `assertInHTML` a radsej nevyberies z `input` divu `input` element a nepozries mu atributy rovnako ako ostatnym HTML elementom?

Rovnako v dalsich testoch nizsie.

Reply:
Zdalo sa mi to prehladnejsie kontrolovat tento `input` ako celok. Trocha som to prerobil, nahradil som `find_class` metodou `find`.

Reply:
@viliambalaz Ano ten `find` cez vseobecny X-path vyzera najlepsie.